scripts/taginfo-project/generate-taginfo-project-file.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+# -------------------------------------------------------------------------------
+#
+# This code generates a taginfo project list file (see more https://wiki.openstreetmap.org/wiki/Taginfo/Projects )
+#
+### Attribution & Disclaimer:
+# This code is based on :
+# Paul Norman code : https://github.com/osmlab/osm2pgsql_taginfo
+# Sven Geggus code : https://github.com/giggls/openstreetmap-carto-de/blob/master/views_osmde/generate_taginfo.py
+# -------------------------------------------------------
+
+import argparse
+import json
+import os
+import re
+import yaml
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-v", "--verbose", help="Print debug info",action="store_true")
+args = parser.parse_args()
+
+
+# -------------------------  parameters ----------------------------
+taginfo = {
+    "data_format": 1,
+    "project": {
+        "name": "OpenStreetMap Carto keys",
+        "description": "Default OpenStreetMap.org style using CartoCSS",
+        "project_url": "https://github.com/gravitystorm/openstreetmap-carto",
+        "contact_name": "openstreetmap-carto maintainers",
+        "contact_email": "openstreetmap-carto (at) gravitystorm (dot) co (dot) uk"
+    },
+    "tags": []
+}
+cwd = os.getcwd()
+
+osm2pgsql_file        = os.path.join(cwd, '..','..', 'openstreetmap-carto.style')
+cartocss_project_file = os.path.join(cwd, '..','..', 'project.mml')
+search_url            = 'https://github.com/gravitystorm/openstreetmap-carto/search?utf8=%E2%9C%93&q='
+
+# output
+taginfo_project_file  = os.path.join(cwd, '..','..', 'taginfo-openstreetmap-carto.json')
+
+
+# ----------------------------------  Examples --------------------
+#  tags @> 'capital=>yes'"]
+#  tags ? 'wetland'"
+#  tags->'wetland'
+#  tags->'leaf_type'
+#  tags @> '"generator:source"=>wind'
+re_tags_one     = re.compile(r"[^a-zA-Z0-9_]tags[^'^)^\[^\]]*'.+?'")
+
+
+# ----------------------------------  Examples --------------------
+#  tags -> ARRAY['wheelchair',ramp:wheelchair']
+#  tags ?& ARRAY['wheelchair',ramp:wheelchair']
+#  tags ?| ARRAY['wheelchair',ramp:wheelchair']
+re_tags_array = re.compile(  r"[^a-zA-Z0-9_]tags\s*[@\?-][>&\|]\s*[aA][rR][rR][aA][yY]\[.+?\]"  )
```

Comment:
Use [re.IGNORECASE](https://docs.python.org/3.1/library/re.html#re.IGNORECASE) as a flag for re.compile to simplify the expression